sound/soc/sof/intel/hda-dsp.c

```diff
@@ -433,6 +433,19 @@ int hda_dsp_runtime_resume(struct snd_sof_dev *sdev)
 	return hda_resume(sdev);
 }
 
+int hda_dsp_runtime_idle(struct snd_sof_dev *sdev)
+{
+	struct hdac_bus *hbus = sof_to_bus(sdev);
+
+	if (hbus->codec_powered) {
+		dev_dbg(sdev->dev, "some codecs still powered (%08X), not idle\n",
```

Comment:
here please change %08X to %lX for I got warning with your patch

Reply:
@RanderWang Thanks! The cast should avoid the warning, but agreed better to solve this without any cast. I'll upload an updated version.